### ISSUE TITLE
JSRuntime.Current is null in Startup.cs

### ISSUE THREAD

**Issue by chanan** [aspnet/Blazor#1746]
I have a library project that needs to do some initialization is the host project's Startup.cs that includes using javascript. This used to work prior to the JS syntax change. Now, however, JSRuntime.Current is null. I created a reproduction of the problem here: https://github.com/chanan/JSRuntimeRepro

**Comment by enetstudio** [comment]
@chanan,
JSRuntime.Current is null because your custom code is executed before it has the chance to 
be initialized... You should follow the execution of the Blazor initialization process to detect how you can insert your code. Perhaps you start with this: https://github.com/aspnet/Blazor/blob/3e062cdefe334be161c8c5f8b3df25283a2fb8d6/src/Microsoft.AspNetCore.Components.Browser/Hosting/WebAssemblyHostBuilder.cs

Hope this helps...

**Comment by chanan** [comment]
@enetstudio Thanks, I did use WebAssemblyHostBuilder as a workaround.

I am wondering if JSRuntime can be initialized prior to call to the IoC. If not, this ticket can be closed.

**Comment by enetstudio** [comment]
Yes, this is a question I'd love to know... Why do you call an issue a ticket ?

**Comment by chanan** [comment]
Force of habit, sorry.

**Comment by SteveSandersonMS** [comment]
TBH I can't immediately think of anything that would strictly stop us from making `JSRuntime.Current` be initalized before the call to `Startup`'s `Configure` method. It *should* be viable in both the server-side and client-side execution models, since by the time we call `Configure`, the whole DI service is prepared and we already have a per-user JS runtime instance.

I'll rename this issue to suggest that as a future enhancement.
